src/vtl.ts

```diff
@@ -418,4 +437,78 @@ export class VTL {
     const __exhaustive: never = node;
     return __exhaustive;
   }
+
+  private mapForEachArgs(call: CallExpr) {
+    const fn = call.args.callbackfn as FunctionExpr;
+    return fn.parameters.map((p) => p.name);
+  }
+
+  private renderMapOrForEach(
+    call: CallExpr,
+    list: string,
+    // Should start with $
+    returnVariable?: string,
+    index?: string,
+    array?: string
+  ) {
+    if (index) {
+      this.add(`#set($${index} = $foreach.index)`);
```

Comment:
Why `$${index}` instead of `${index}`? I have been trying to adopt the convention of having variables always contain the `$`.

Reply:
I didn't change this, index and array were unformatted in your code too. I can update them though.

Reply:
Oh my bad. I changed my mind a few times, left it a bit messy it seems.

If it's easy to do so, please update. No big deal tho.